### PR TITLE
chore: initialising journal

### PR DESCRIPTION
**Pull request by thevaibhav-dixit**

### CODE REVIEW

File: cala-ledger-core-types/src/journal.rs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+use serde::{Deserialize, Serialize};
+
+use super::primitives::*;
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct JournalValues {
+    pub id: JournalId,
+    pub name: String,
+    pub external_id: Option<String>,
```
Comment (thevaibhav-dixit) [inline cala-ledger-core-types/src/journal.rs:9]: in sqlx-ledger we also have a ```version ``` field i was unsure about what it is used for, can add that as well. 
[link](https://github.com/GaloyMoney/sqlx-ledger/blob/main/ledger/src/journal/entity.rs#L12)

Comment (bodymindarts) [reply cala-ledger-core-types/src/journal.rs:9]: That's fine - we will introduce it via `events.len()` - no need for it as a field.

File: cala-ledger/migrations/20231208110808_cala_ledger_setup.sql [hunk 2]
```diff
@@ -29,6 +29,28 @@ CREATE TABLE cala_account_events (
   FOREIGN KEY (connection_id, id) REFERENCES cala_accounts(connection_id, id)
 );
 
+CREATE TABLE cala_journals (
+  connection_id UUID NOT NULL DEFAULT '00000000-0000-0000-0000-000000000000', 
+  id UUID PRIMARY KEY,
+  name VARCHAR NOT NULL, 
+  external_id VARCHAR NOT NULL,
```
Comment (bodymindarts) [inline cala-ledger/migrations/20231208110808_cala_ledger_setup.sql:36]: external_id is optional in Cala

File: cala-ledger/migrations/20231208110808_cala_ledger_setup.sql [hunk 3]
```diff
@@ -29,6 +29,28 @@ CREATE TABLE cala_account_events (
   FOREIGN KEY (connection_id, id) REFERENCES cala_accounts(connection_id, id)
 );
 
+CREATE TABLE cala_journals (
+  connection_id UUID NOT NULL DEFAULT '00000000-0000-0000-0000-000000000000', 
+  id UUID PRIMARY KEY,
```
Comment (bodymindarts) [inline cala-ledger/migrations/20231208110808_cala_ledger_setup.sql:34]: id is not a primary key... rather (connection_id, id) is unique.

File: cala-ledger/migrations/20231208110808_cala_ledger_setup.sql [hunk 4]
```diff
@@ -29,6 +29,28 @@ CREATE TABLE cala_account_events (
...
+);
+
+CREATE TABLE cala_journal_events (
+  connection_id UUID NOT NULL DEFAULT '00000000-0000-0000-0000-000000000000',
+  id UUID NOT NULL,
+  sequence INT NOT NULL,
+  event_type VARCHAR NOT NULL,
+  event JSONB NOT NULL,
+  recorded_at TIMESTAMPTZ NOT NULL DEFAULT NOW(),
+  UNIQUE(id, sequence),
```
Comment (bodymindarts) [inline cala-ledger/migrations/20231208110808_cala_ledger_setup.sql:49]: unique must include connection_id

File: cala-ledger/migrations/20231208110808_cala_ledger_setup.sql [hunk 5]
```diff
@@ -29,6 +29,28 @@ CREATE TABLE cala_account_events (
...
 
+CREATE TABLE cala_journals (
+  connection_id UUID NOT NULL DEFAULT '00000000-0000-0000-0000-000000000000', 
+  id UUID PRIMARY KEY,
+  name VARCHAR NOT NULL, 
+  external_id VARCHAR NOT NULL,
+  created_at TIMESTAMPTZ NOT NULL DEFAULT NOW(),
+  UNIQUE(connection_id, id),
+  UNIQUE(connection_id, name)
+);
```
Comment (bodymindarts) [inline cala-ledger/migrations/20231208110808_cala_ledger_setup.sql:40]: missing indexes - look at account tables.

File: cala-ledger/src/journal/repo.rs [hunk 6]
```diff
@@ -0,0 +1,33 @@
+use sqlx::{PgPool, Postgres, Transaction};
+
+use super::{entity::*, error::*};
+use crate::entity::*;
+
+#[derive(Debug, Clone)]
+pub(super) struct JournalRepo {
+    pool: PgPool,
+}
```
Comment (bodymindarts) [inline cala-ledger/src/journal/repo.rs:9]: newline.

**Review by bodymindarts — COMMENTED** [review]
you are missing the use case in the journal/mod.rs file that forwards to outbox - look at accounts.

File: cala-ledger/src/journal/mod.rs [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+use sqlx::PgPool;
+use tracing::instrument;
+
+use crate::outbox::*;
+
+pub use entity::*;
+use error::*;
+use repo::*;
+
+/// Service for working with `Account` entities.
```
Comment (bodymindarts) [inline cala-ledger/src/journal/mod.rs:14]: `Journal` entities

**Review by bodymindarts — APPROVED** [review]
Alright looks good! Next cala-nodejs layer.

**Pull request merged**